codebundles/k8s-kubectl-namespace-healthcheck/runbook.robot

```diff
@@ -31,53 +31,84 @@ Suite Initialization
     ...    example=Kubernetes
+    ...    default=Kubernetes
     ${ERROR_PATTERN}=    RW.Core.Import User Variable    ERROR_PATTERN
     ...    type=string
     ...    description=The error pattern to use when grep-ing logs.
     ...    pattern=\w*
     ...    example=(Error|Exception)
     ...    default=(Error|Exception)
+    ${EVENT_AGE}=    RW.Core.Import User Variable    EVENT_AGE
+    ...    type=string
+    ...    description=The time window in minutes as to when the event was last seen.
+    ...    pattern=((\d+?)m)?
+    ...    example=30m
+    ...    default=30m
     ${binary_name}=    RW.K8s.Get Binary Name    ${DISTRIBUTION}
+    ${kinds_list}=    RW.utils.Csv To List    ${RESOURCE_KINDS}
+    ${RESOURCE_KINDS}=    RW.utils.Remove Spaces   ${RESOURCE_KINDS}
     Set Suite Variable    ${binary_name}    ${binary_name}
+    Set Suite Variable    ${kinds_list}    ${kinds_list}
+    Set Suite Variable    ${kubeconfig}    ${kubeconfig}
+    Set Suite Variable    ${kubectl}    ${kubectl}
+    Set Suite Variable    ${CONTEXT}    ${CONTEXT}
+    Set Suite Variable    ${NAMESPACE}    ${NAMESPACE}
+    Set Suite Variable    ${ERROR_PATTERN}    ${ERROR_PATTERN}
+    Set Suite Variable    ${RESOURCE_KINDS}    ${RESOURCE_KINDS}
+    Set Suite Variable    ${EVENT_AGE}    ${EVENT_AGE}
 
 *** Tasks ***
-Troubleshoot Namespace
-    ${stdout}=    RW.K8s.Shell
-    ...    cmd=kubectl api-resources --verbs=list --namespaced --context=${CONTEXT} --namespace=${NAMESPACE} -o name
-    ...    target_service=${kubectl}
-    ...    kubeconfig=${kubeconfig}
-    ${resource_types}=    RW.Utils.Stdout To List    stdout=${stdout}
-    ${object_names}=    RW.K8s.Loop Template Shell
-    ...    items=${resource_types}
-    ...    cmd=kubectl get {item} --no-headers --show-kind --ignore-not-found --context=${CONTEXT} --namespace=${NAMESPACE} -o name
-    ...    target_service=${kubectl}
-    ...    kubeconfig=${kubeconfig}
-    ...    newline_as_separate=True
-    ${ts_results}=    RW.K8s.Check Namespace Objects
-    ...    k8s_object_names=${object_names}
+Trace Namespace Errors
+    ${error_results}=    RW.K8s.Trace Namespace Errors
     ...    context=${CONTEXT}
     ...    namespace=${NAMESPACE}
     ...    target_service=${kubectl}
     ...    kubeconfig=${kubeconfig}
+    ...    error_pattern=${ERROR_PATTERN}
+    ...    event_age=${EVENT_AGE}
+    ...    binary_name=${binary_name}
     ${history}=    RW.K8s.Pop Shell History
     ${history}=    RW.Utils.List To String    data_list=${history}
-    RW.Core.Add Pre To Report    ${ts_results}
+    RW.Core.Add Pre To Report    Summary of errors traced in namespace: ${NAMESPACE}
+    RW.Core.Add Pre To Report    ${error_results}
     RW.Core.Add Pre To Report    Commands Used:\n${history}
 
-Find Namespace Errors
-    ${error_results}=    RW.K8s.Check Namespace Errors
-    ...    context=${CONTEXT}
-    ...    namespace=${NAMESPACE}
+
+*** Tasks ***
+Fetch Unready Pods
+    ${unreadypods_results}=    RW.K8s.Shell
```

Comment:
I one of the template generators, at least for an SLI, I was doing this with jmespath and a search filter like this: 
`status.replicas != status.availableReplicas && metadata.name == [namspace ref]`

Not saying it needs to be used here as this is a nice clean task;  just sharing in case its more performant in larger queries or you already have the data collected and want a different manipulation structure than the kubectl output.